bootcamp_project/Module 10/Promises/BookAPISimulation.ts

```diff
@@ -0,0 +1,100 @@
+class UserData {
+
+    userID: number;
+    userName: string;
+
+    constructor(userID: number, userName: string) {
+        this.userID = userID;
+        this.userName = userName;
+    }
+
+}
+
+class BookData {
+
+    bookID: number;
+    bookName: string;
+
+
+    constructor(bookID: number, bookName: string) {
+        this.bookID = bookID;
+        this.bookName = bookName;
+    }
+
+}
+
+function fetchBook(bookID: number): Promise<BookData> {
+
+    const books: BookData[] = [
+        { bookID: 1, bookName: "Alice in Fantasyland" },
+        { bookID: 2, bookName: "Draco Malfoy and the Vanishing Apple" },
+        { bookID: 3, bookName: "Game of Crowns" },
+        { bookID: 4, bookName: "Duke of the Earrings" },
+        { bookID: 5, bookName: "Animal Ranch" }
```

Comment:
These are being initialized as object literals. This is... Interesting because they meet the interface criteria defined by the bookdata class since they have the 2 required fields, but in this case, the constructor is never being executed and that **could be** an issue.

I would suggest:
```suggestion
        new BookData(1, "Alice in Fantasyland"),
        ...
```